### PR TITLE
github-pull-request-make: control stress parallelism

### PR DESCRIPTION
**Pull request by andreimatei**
A new test struggled to get past its stressrace (the stress applied to
new/changed tests), because the machine seemed overloaded, presumably
because we're not limiting the parallelism of the stress (which results
in parallelism = #cores). It's surprising that we didn't limit the
parallelism of individual tests already; when stressing whole packages
for example we use -p 1. When we run testrace
on all the packages, we use -p 2.

This patch uses -p 16 for stress, and -p 8 for stressrace.

Release note: None

### CODE REVIEW

**Comment by andreimatei** [comment]
cc @ajstorm

File: pkg/cmd/github-pull-request-make/main.go [hunk 1]
```diff
@@ -273,14 +273,19 @@ func main() {
 					tests = "(" + strings.Join(pkg.tests, "$$|") + "$$)"
 				}
 
+				parallelism := 16
```
Comment (tbg) [inline pkg/cmd/github-pull-request-make/main.go:276]: Add a comment. Something like

> The `stress -p` flag defaults to the number of CPUs, which in CI is usually 16. This is very aggressive and can cause tests to fail. Under nightly stress, we usually use 4 or 2, so run with 8 here to make sure the test becomes an obvious candidate for skipping under race before it has to deal with the nightlies.

**Review by tbg — APPROVED** [review]

**Review by andreimatei — COMMENTED** [review]
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/61944#-:-MVqX7Np0x5kuvpFdS21:b-ydi9ay)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jlinder and @tbg)

___
*[pkg/cmd/github-pull-request-make/main.go, line 276 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/61944#-MVcNWTc1IehpLwzOuIP-r1-276:-MVqX6RSDkI7Q7v0ohmj:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9c9dc60507b54b8a938b0d9db76c8f459e25f79/pkg/cmd/github-pull-request-make/main.go#L276)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

Add a comment. Something like

> The `stress -p` flag defaults to the number of CPUs, which in CI is usually 16. This is very aggressive and can cause tests to fail. Under nightly stress, we usually use 4 or 2, so run with 8 here to make sure the test becomes an obvious candidate for skipping under race before it has to deal with the nightlies.
</blockquote></details>

done

**Comment by ajstorm** [comment]
@andreimatei any opposition to me backporting this to 21.1?  I have a new test case in master that I'd like to backport, but the lack of this PR in the 21.1 release is making that difficult.

https://teamcity.cockroachdb.com/viewLog.html?buildId=2848600

**Comment by andreimatei** [comment]
go for it

**Pull request merged**